KY-032_AvoidanceTracking/KY-032_AvoidanceTracking.ino

```diff
@@ -1,25 +1,99 @@
+ *    David Schmidt (davids@inventr.io)
+ *    Frank Visco (bittoe@yahoo.com)
+ *
+.-------------------------------.
+> Avoidance tracker Calibration <
+:---------------------------------------------------------------------------------------:
+| To get this to work you have to adjust both the output and input potentiometers (POT).
+|
+| The POT on the right, which sits just below the EN jumper, adjusts the LED output.
+| -Turn this clockwise (to the right) as far as it will go. 
+|
+| The POT on the left adjusts sensor sensitivity.
+| -Turn this as far counterclockwise (to the left) as it will go.
+| -Place your hand or some other flat object about 4″ (~10cm) above the sensor.
+| -SLOWLY turn the POT clockwise (right) until the LED on the PCB comes on and the serial output says “obstacle in the way”
+|
+| There is a small range for detection but With a little finesse, gentle nudging, and tweaking
+| of the POT you should be able to get a detection range between 0 and ~10″ (~25cm).
+| You really have to play with the detection sensitivity to get it to work reliably and consistently
+:---------------------------------------------------------------------------------------:
+ *
+ * This project uses: 
+ * One Digital pin to INPUT (object detection).
+ * One *optional* digital pin to OUTPUT (enable/disable sensor)
+ * On the Hero (Arduino Uno compatible) we *could* use: D0-D13, A0-A5.
+ * Skip: A0-A5 (save for Analog),
+ *       D0/D1 (used by USB),
+ *       D2/D3 (save for interrupts),
+ *       D13 (used by LED_BUILTIN and SPI Clock),
+ *       D5, D6, D9, D10 and D11 (save for PWM)
+ *       D11 (SPI MOSI)
+ *       D12 (SPI MISO)
+ * Recommended for fewest conflicts:
+ *    D4, D7 or D8
  */
+const uint8_t KY_032_PIN_OUT = 8;   // set sensor output to a digitial pin 
+// If you want the ability to enable and disable the sensor manually 
+// you must remove the EN jumper and 
+// Assign the EN pin to a digital pin:
+//    const uint8_t KY_032_PIN_EN = ; 
+// Set the pin mode
+//    pinMode(KY-032_PIN_EN, OUTPUT);
+// Read the value of the OUTPUT pin
+//    int KY_032_enable = digitalRead(KY_032_PIN_EN);
 
 void setup() {
-  Serial.begin(9600);
-  pinMode(8, INPUT);  // set pin as input
+Serial.begin(9600);
+pinMode(KY_032_PIN_OUT, INPUT);// set pin as input
 }
 
 void loop() {
-  //Written for Robojax on Dec 28, 2017
-  int detect = digitalRead(8);  // read obstacle status and store it into "detect"
-  if (detect == LOW) {
-
+  int KY_032_state = digitalRead(KY_032_PIN_OUT);  // read sensor state and store it into "KY_032_state"
+  
+  if (KY_032_state == LOW) {                       // This could also be written if(!KY_032_state){
     Serial.println("Obstacle in the way");
-  } else {
 
-    Serial.println("clear");
+  } else {
+    Serial.println("The path is clear");
   }
-  delay(500);
+  Serial.print("\nKY-032 output = "); // Print the digital output of the sensor for debugging purposes
+  Serial.println(KY_032_state);  
```

Comment:
Do we need to print the actual state output here?  And if we don't, then we also don't need to save the data in a variable.  We can just do the digitalRead() in the "if" statement.

Reply:
I re-worked this for your consideration.
I eliminated the KY_32_state variable as suggested
On line 87 I commented-in an alternate if statement because I figure some people may be unfamiliar with this alternative
I included the Serial print because I used this info in most of my sketches to see exactly what the sensor was writing and it helped me learn about what was really going on. But I moved the Serial.print() to a comment section 

There are two things I forgot to do:
1) Explain the LEDs on the sensor
2) Use the Hero LED for detection verification